hoomd/mpcd/CollisionMethod.cc

```diff
@@ -48,15 +78,945 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
+    ArrayHandle<Scalar4> h_orientation(m_pdata->getOrientationArray(),
+                                       access_location::host,
+                                       access_mode::read);
+    ArrayHandle<Scalar3> h_inertia(m_pdata->getMomentsOfInertiaArray(),
+                                   access_location::host,
+                                   access_mode::read);
+    ArrayHandle<unsigned int> h_rigid_center(m_rigid_bodies->getRigidCenters(),
+                                             access_location::host,
+                                             access_mode::read);
+    // add accumulated momentum to the central particle
+    unsigned int num_centers = m_rigid_bodies->getNLocal();
+    for (unsigned int idx = 0; idx < num_centers; ++idx)
+        {
+        // get index of central particle
+        const unsigned int central_idx = h_rigid_center.data[idx];
+
+        // get accumulated momentum for particle
+        const Scalar3 linmom_accum(h_linmom_accum.data[central_idx]);
+        const vec3<Scalar> angmom_accum(h_angmom_accum.data[central_idx]);
+
+        // compute and store new velocity
+        Scalar4 vel_mass = h_velocity.data[central_idx];
+        const Scalar mass = vel_mass.w;
+        if (mass > 0)
+            {
+            vel_mass.x += linmom_accum.x / mass;
+            vel_mass.y += linmom_accum.y / mass;
+            vel_mass.z += linmom_accum.z / mass;
+            h_velocity.data[central_idx] = vel_mass;
+            }
+
+        // compute new angular momentum
+        quat<Scalar> angmom(h_angmom.data[central_idx]);
+        const quat<Scalar> orientation(h_orientation.data[central_idx]);
+
+        // convert angular momentum to quaternion and update
+        const vec3<Scalar> inertia(h_inertia.data[central_idx]);
+        vec3<Scalar> angmom_change_body = rotate(conj(orientation), angmom_accum);
+        if (inertia.x == Scalar(0))
+            {
+            angmom_change_body.x = Scalar(0);
+            }
+
+        if (inertia.y == Scalar(0))
+            {
+            angmom_change_body.y = Scalar(0);
+            }
+
+        if (inertia.z == Scalar(0))
+            {
+            angmom_change_body.z = Scalar(0);
+            }
+        angmom += Scalar(2.0) * orientation * quat(0.0, angmom_change_body);
+
+        // save update
+        h_angmom.data[central_idx] = quat_to_scalar4(angmom);
+        }
+    }
+
+#ifdef ENABLE_HIP
+//! Create autotuners
+void mpcd::CollisionMethod::checkRigidAutotuners()
+    {
+    const bool rigid_body_collision = m_embed_group && m_rigid_bodies;
+    std::vector<std::shared_ptr<AutotunerBase>> new_autotuners;
+    new_autotuners.insert(new_autotuners.end(),
+                          {m_drawrandvec_tuner,
+                           m_netvelo_tuner,
+                           m_applyrandvec_tuner,
+                           m_store_tuner,
+                           m_accumulate_tuner,
+                           m_transfer_tuner});
```

Comment:
I think you can just do brace initialization here. Also, suggest to name `rigid_autotuners` since they aren't always new.
```suggestion
    std::vector<std::shared_ptr<AutotunerBase>> rigid_autotuners {m_drawrandvec_tuner,
                           m_netvelo_tuner,
                           m_applyrandvec_tuner,
                           m_store_tuner,
                           m_accumulate_tuner,
                           m_transfer_tuner};
```